src/lib/cfa/sudoers.rb

```diff
@@ -0,0 +1,237 @@
+      end
+    end
+
+    # Constructor
+    #
+    # @param file_handler [#read,#write] something able to read/write a string (like File)
+    #
+    # @see CFA::BaseModel#initialize
+    def initialize(file_path: DEFAULT_PATH, file_handler: nil)
+      super(PARSER, file_path, file_handler: file_handler)
+    end
+
+    # Read and prepare the data
+    #
+    # @see CFA::BaseModel#initialize
+    # @see #fix_collection_names
+    # @see #set_start_placer
+    def load
+      super
+
+      fix_collection_names(data)
+      @start_placer = set_start_placer
+    end
+
+    # Return all users specifications
+    #
+    # @return [Array<Hash>]
+    def users_specs
+      current_users.reduce([]) do |specs, user|
+        spec = {}
+        values = user[:value]
+
+        spec["user"] = values["user"]
+        spec["host"] = values["host_group"]["host"]
+        spec["commands"] = values["host_group"].collection("command").map do |command|
+          command.is_a?(AugeasTreeValue) ? command.value : command
+        end
+        spec["run_as"] = values["host_group"]["command[]"]["runas_user"]
+        spec["tag"] = values["host_group"]["command[]"]["tag"]
+
+        specs << spec.reject { |_, v| v.nil? }
+      end
+    end
+
+    # Remove all users specifications
+    def clean_users
+      data.delete(USERS_MATCHER)
+    end
+
+    # Add a new basic user specification
+    #
+    #
+    # A user specification determines which commands a user may run (and as what user) on specified
+    # hosts (who may run what, for short).
+    #
+    # Its basic structure is “who where = (as_whom) what”.
+    #
+    # E.g., with
+    #
+    #   root            ALL = (ALL) ALL
+    #   %wheel          ALL = (ALL) ALL
+    #
+    #   We let root and any user in group wheel run any command on any host as any user.
+    #
+    # See the sudoers manpage for more details.
+    #
+    # @param user_spec [Hash] a user specification
+    # @option user_spec [String] "user" who can run the command
+    # @option user_spec [String] "host" where the user can run the command
+    # @option user_spec [Array<String>] "commands" commands that the user can run
+    # @option user_spec [String] "run_as" as whom the commands will be run
+    # @option user_spec [String] "tag" the tag associated to the command (e.g. NOPASSWD)
+    def add_user(user_spec)
+      user = AugeasTree.new
+      host_group = AugeasTree.new
+
+      host_group.add("host", user_spec["host"])
+
+      commands = user_spec["commands"].reduce([]) do |commands, command|
+        augeas_command = AugeasTree.new
+        host_group.add("command[]", AugeasTreeValue.new(augeas_command, command))
+        commands << augeas_command
+      end
+
+      # For the time being, "runas_user" and "tag" options are added only to the first command,
+      # which means that they apply to all commands.
+      run_as = user_spec["run_as"].to_s.strip
+      tag = user_spec["tag"].to_s.strip
```

Comment:
is it valid to have here empty string?

Reply:
Nope. That's why in "run_ass" and "tag" are being added `unless` they are empty in the following lines.

I did `to_s.strip` to simplify the below condition to `unless run_as.empty?` instead to use `unless run_as.nil? || run_as.empty?`